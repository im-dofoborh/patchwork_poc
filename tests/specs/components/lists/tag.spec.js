import { mount, createLocalVue } from '@vue/test-utils';
import VueRouter from 'vue-router';
import TagList from '~/components/lists/Tag';

describe('TagList component', () => {
  let router = null;
  let localVue = null;
  let wrapper = null;

  beforeEach(() => {
    localVue = createLocalVue();
    localVue.component('tag-list', TagList);
    localVue.use(VueRouter);
    router = new VueRouter();
    wrapper = mount(TagList, {
      localVue,
      router,
    });
  })

  it('renders default variant', () => {
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('renders with multiple tags', () => {
    wrapper.setProps({
      tags: [
        'foo',
        'bar',
        'baz',
      ]
    })
    expect(wrapper.html()).toMatchSnapshot();
  });
});
