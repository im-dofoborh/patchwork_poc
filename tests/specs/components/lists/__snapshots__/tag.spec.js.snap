// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TagList component renders default variant 1`] = `
<div class="sg-tag-list">
  <h3 class="sg-tag-list__title">Tags</h3>
  <ul class="sg-tag-list__list">
    <li></li>
  </ul>
</div>
`;

exports[`TagList component renders with multiple tags 1`] = `
<div class="sg-tag-list">
  <h3 class="sg-tag-list__title">Tags</h3>
  <ul class="sg-tag-list__list">
    <li><a href="#/components/list?search=foo" class=""><span class="sg-tag-list__item">
          foo
        </span> <span class="sg-tag-list__seperator body-copy-small">
          |
        </span></a><a href="#/components/list?search=bar" class=""><span class="sg-tag-list__item">
          bar
        </span> <span class="sg-tag-list__seperator body-copy-small">
          |
        </span></a><a href="#/components/list?search=baz" class=""><span class="sg-tag-list__item">
          baz
        </span>
        <!----></a></li>
  </ul>
</div>
`;
